comp/core/status/statusimpl/status_api_endpoints_test.go

```diff
@@ -0,0 +1,128 @@
+// Unless explicitly stated otherwise all files in this repository are licensed
+// under the Apache License Version 2.0.
+// This product includes software developed at Datadog (https://www.datadoghq.com/).
+// Copyright 2023-present Datadog, Inc.
+
+// Package statusimpl implements the status component interface
+package statusimpl
+
+import (
+	"encoding/json"
+	"net/http"
+	"net/http/httptest"
+	"os"
+	"testing"
+	"time"
+
+	"go.uber.org/fx"
+
+	"github.com/gorilla/mux"
+
+	"github.com/stretchr/testify/require"
+
+	"github.com/DataDog/datadog-agent/comp/core/config"
+	"github.com/DataDog/datadog-agent/comp/core/log/logimpl"
+	"github.com/DataDog/datadog-agent/comp/core/status"
+	pkgconfigsetup "github.com/DataDog/datadog-agent/pkg/config/setup"
+	"github.com/DataDog/datadog-agent/pkg/util/fxutil"
+)
+
+func getTestComp(t *testing.T) provides {
+	deps := fxutil.Test[dependencies](t, fx.Options(
+		config.MockModule(),
+		logimpl.MockModule(),
+		fx.Supply(
+			agentParams,
+			status.NewInformationProvider(mockProvider{
```

Comment:
Should we add a header provider to ensure we cover all cases 😄

Reply:
Good idea, done in 5020fd93fb0946db9dcf7cee69c71d80ffeb5b2f.